core/test-output-stream.ts

```diff
@@ -69,11 +69,12 @@ export class TestOutputStream extends ReadableStream {
 
       this._writeOut(`not ok ${testId} ${description}\n`);
 
-      if (error instanceof MatchError === false) {
-         error = new MatchError("the test threw an error", "the test to run", "Test threw " + (<any>error.constructor).name + " with message \"" + error.message + "\"");
+      // if it's a match error then log it properly, otherwise log it as unhandled
+      if (error instanceof MatchError) {
+          this._writeMatchErrorOutput(<MatchError>error);
```

Comment:
do you need the type guard here, I thought that TypeScript will have an implicit one if it's inside a typed block like the above if statement?